lib/ansible/modules/network/cloudengine/ce_vrf_interface.py

```diff
@@ -0,0 +1,517 @@
+    type: dict
+    sample: {
+                "vpn_interface": "40GE2/0/17",
+                "vrf": null
+            }
+end_state:
+    description: k/v pairs of end attributes on the interface
+    returned: always
+    type: dict or null
+    sample: {
+                "vpn_interface": "40GE2/0/17",
+                "vrf": "jss"
+            }
+updates:
+    description: command list sent to the device
+    returned: always
+    type: list
+    sample: [
+                "ip binding vpn-instance jss",
+            ]
+changed:
+    description: check to see if a change was made on the device
+    returned: always
+    type: boolean
+    sample: true
+'''
+
+
+from xml.etree import ElementTree
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ce import ce_argument_spec, get_nc_config, set_nc_config
+
+CE_NC_GET_VRF = """
+<filter type="subtree">
+  <l3vpn xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <l3vpncomm>
+      <l3vpnInstances>
+        <l3vpnInstance>
+          <vrfName>%s</vrfName>
+        </l3vpnInstance>
+      </l3vpnInstances>
+    </l3vpncomm>
+  </l3vpn>
+</filter>
+"""
+
+CE_NC_GET_VRF_INTERFACE = """
+<filter type="subtree">
+  <l3vpn xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <l3vpncomm>
+      <l3vpnInstances>
+        <l3vpnInstance>
+          <vrfName></vrfName>
+          <l3vpnIfs>
+            <l3vpnIf>
+              <ifName></ifName>
+            </l3vpnIf>
+          </l3vpnIfs>
+        </l3vpnInstance>
+      </l3vpnInstances>
+    </l3vpncomm>
+  </l3vpn>
+</filter>
+"""
+
+CE_NC_MERGE_VRF_INTERFACE = """
+<config>
+  <l3vpn xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <l3vpncomm>
+      <l3vpnInstances>
+        <l3vpnInstance>
+          <vrfName>%s</vrfName>
+          <l3vpnIfs>
+            <l3vpnIf operation="merge">
+              <ifName>%s</ifName>
+            </l3vpnIf>
+          </l3vpnIfs>
+        </l3vpnInstance>
+      </l3vpnInstances>
+    </l3vpncomm>
+  </l3vpn>
+</config>
+"""
+
+CE_NC_GET_INTF = """
+<filter type="subtree">
+  <ifm xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <interfaces>
+      <interface>
+        <ifName>%s</ifName>
+        <isL2SwitchPort></isL2SwitchPort>
+      </interface>
+    </interfaces>
+  </ifm>
+</filter>
+"""
+
+CE_NC_DEL_INTF_VPN = """
+<config>
+  <l3vpn xmlns="http://www.huawei.com/netconf/vrp" content-version="1.0" format-version="1.0">
+    <l3vpncomm>
+      <l3vpnInstances>
+        <l3vpnInstance>
+          <vrfName>%s</vrfName>
+          <l3vpnIfs>
+            <l3vpnIf operation="delete">
+              <ifName>%s</ifName>
+            </l3vpnIf>
+          </l3vpnIfs>
+        </l3vpnInstance>
+      </l3vpnInstances>
+    </l3vpncomm>
+  </l3vpn>
+</config>
+"""
+
+
+def get_interface_type(interface):
```

Comment:
If this is needed by other modules it may be useful to move this into `module_utils`.